src/Core/Application/Registry/AbstractRegistry.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Core\Application\Registry;
+
+use InvalidArgumentException;
+
+/**
+ * @template T of \SprykerSdk\Sdk\Core\Application\Registry\RegistryItemInterface
+ * @implements \SprykerSdk\Sdk\Core\Application\Registry\RegistryInterface<T>
+ */
+abstract class AbstractRegistry implements RegistryInterface
+{
+    /**
+     * @var bool
+     */
+    protected bool $isLoaded = false;
+
+    /**
+     * @var iterable<T>
+     */
+    protected iterable $iterableItems;
+
+    /**
+     * @var array<string, T>
+     */
+    protected array $items;
```

Comment:
in this case should be static with isLoaded.
for each extended class we will upload this data

Reply:
DI should resolve instance management